### ISSUE TITLE
This library does not conform to RN standards

### ISSUE THREAD

**Issue by billnbell** [forcedotcom/SalesforceMobileSDK-ReactNative#325]
There are several real issues with this library. 

1. First the RN standard is to create a package that loads in Android and IOS - not doing it in a CLI command
2. The RN 70.* new architecture and removing the RN Bridging mode needs to happen

Deploying to NPM so that we can `yarn add react-native-force` and then `yarn ios` / `yarn android` should be ALL you ever need.

Can we get some help forking this and fixing it to conform to how everyone else does it?

**Comment by wmathurin** [comment]
We have it in our backlog to revisit how we do our react native templates and bridging, but no time set yet for getting to that work.

**Comment by JFoxUK** [comment]
This is an absolute must and is one of the reasons why we stay clear from using the Salesforce SDK most of the time.

**Comment by billnbell** [comment]
Yeah if you just look at this repo that is a good example of how to do it.

I converted this to work with RN Bridging and this compiles without doing any injection into Xcode or Android Studio files.

https://github.com/billnbell/react-native-sunshine-conversations

1. Create a pod spec [react-native-smooch.podspec](https://github.com/billnbell/react-native-sunshine-conversations/blob/master/react-native-smooch.podspec)
2. Create build.gradle with dependencies https://github.com/billnbell/react-native-sunshine-conversations/blob/master/android/build.gradle

**Comment by billnbell** [comment]
Or just copy your competitor's setup. https://github.com/FormidableLabs/react-native-app-auth

**Comment by JFoxUK** [comment]
@wmathurin @bbirman is this on the roadmap?

**Comment by hmkcfs** [comment]
@wmathurin @bbirman , React Native team officially suggested to use frameworks like exp. When will you guys support Expo for Salesforce mobile SDK

https://reactnative.dev/blog/2024/06/25/use-a-framework-to-build-react-native-apps

**Comment by wmathurin** [comment]
The next version of Mobile SDK (14.0) due this summer will be on the new architecture and on the latest version of React Native   (or whatever was the latest version is a month or so before we actually cut the release). We will also move the Android bridges to the SalesforceMobileSDK-ReactNative repo (alongside the iOS ones) to facilitate auto-linking. Some of the changes are already in dev so you can have a look there.
